clientd3d/music.c

```diff
@@ -1,853 +1,257 @@
-      debug(("can't MCI open new song \n"));
-      mciSendCommand(MCI_ALL_DEVICE_ID, MCI_CLOSE, 0, 0);
-      playing_music = false;
-      playing_midi = false;
-      isMusicPaused = false;
-      midi_element = 0;
-      midi_bg_music_element = 0;
-      return dwReturn;
-   }
-   // Save element ID and set flag
-   switch (musicType) {
-      case BACKGROUND_MUSIC:
-         midi_bg_music_element = mciOpenParms.wDeviceID;
-         playing_music = true;
-         debug(("midi_bg_music_element = %d\n", midi_bg_music_element));
-         break;
-      case GAMEPLAY_MUSIC:
-         midi_element = mciOpenParms.wDeviceID;
-         playing_midi = true;
-         debug(("midi element = %d\n", midi_element));
-         break;
-      default:
-         debug(("OpenMidiFile: Invalid musicType, type= %d\n", musicType));
-         return 1;
-   }
 
-   return dwReturn;
-}
 /******************************************************************************/
 /*
- * PlayMidiFile:  Play given MIDI file and notify given window when done.
- *   Returns 0 if successful, MCI error code otherwise.
- *   Only used by music files associated with gameplay actions
- *   and never background music, even if the background music is a midi file.
+ * PlayMidiFile: Legacy compatibility shim for server protocol BP_PLAY_MIDI.
+ * Maps .mid/.midi/.mp3 extensions to .ogg and plays via OpenAL.
  */
 DWORD PlayMidiFile(HWND hWndNotify, char *fname)
 {
-   if (!has_midi)
-      return 0;
-
-#ifdef M59_MSS
-	// If a sample was playing, end it
-	if( AIL_sample_status( hseqImmediate ) != SMP_DONE )
-		AIL_end_sample( hseqImmediate );
-
-	// free memory from previous MIDI file
-	if( pMIDIImmediate )
-		AIL_mem_free_lock( pMIDIImmediate );
-
-	// First try MP3 file
-	char *ext = strstr( _strlwr( fname ), ".mid" );
-	if( ext != NULL )
-		strcpy( ext, ".mp3" );
-
-	// load the file
-	pMIDIImmediate = (BYTE *) AIL_file_read( fname, NULL );
-	if( !pMIDIImmediate )
-	{
-      // Next try xmi file
-      ext = strstr(fname, ".mp3" );
-      if( ext != NULL )
-         strcpy( ext, ".xmi" );
-      
-      pMIDIImmediate = (BYTE *) AIL_file_read( fname, NULL );
-      if( !pMIDIImmediate )
-      {
-         debug(( "Failed to load music file %s.\n", fname ));
-         return 0;
-      }
-	}
-
-	// Initialize the sample
-	if (!AIL_set_named_sample_file(hseqImmediate, fname, pMIDIImmediate,
-                                  AIL_file_size(fname), 0))
-	{
-		debug(( "Failed to init music file.\n" ));
-		return 0;
-	}
-   // Loop the MIDI until game actions stop it
-	AIL_set_sample_loop_count( hseqImmediate, 0 );
-   // Set volume
-   float vol = ((float) config.music_volume) / CONFIG_MAX_VOLUME;
-	AIL_set_sample_volume_levels(hseqImmediate, vol, vol );
-
-	// start playing
-	AIL_start_sample(hseqImmediate);
-
-	debug(( "Playing midi music file %s. LatestMusic=%d \n", fname, latest_music ));
-	playing_midi = true;
-	return 0;
-#else
+   (void)hWndNotify;
+   if (!fname) return 1;
```

Comment:
Not sure what these return values mean, and the documentation is gone.  Could maybe return a bool now?